metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1389,8 +1408,72 @@ class MetalsLanguageServer(
           indexWorkspaceSources()
         }.asJavaObject
       case ServerCommands.RestartBuildServer() =>
-        bloopServers.shutdownServer()
+        bspSession.foreach { session =>
```

Comment:
Could we move the logic to a separate method?

Reply:
Yea for sure, I moved this out now.